src/index.ts

```diff
@@ -16,57 +20,79 @@ async function getFiles(dir: string): Promise<string[]> {
       return dirent.isDirectory() ? getFiles(res) : res;
     })
   );
+
   return Array.prototype
     .concat(...files)
     .filter((file) => path.extname(file) === ".svg");
 }
 
+const parseToJSXComponent = (
+  filePath: string,
+  outputPath: string,
+  callback?: () => any
+) => {
+  const spinner = ora("generating components").start();
+
+  if (!filePath) {
+    spinner.stop();
+    process.exit(0);
+  }
+
+  const componentName = _upperFirst(
+    _camelCase(path.basename(filePath, ".svg"))
+  );
+
+  parser(filePath, componentName).then((text) => {
+    fs.mkdirSync(outputPath, { recursive: true });
+    fs.writeFile(
+      path.join(outputPath, componentName + ".tsx"),
+      text,
+      {
+        encoding: "utf-8",
+      },
+      (error) => {
+        if (error) {
+          console.warn(`error converting: ${filePath}`);
+        } else {
+          spinner.text = `${componentName + ".tsx"} generated`;
+          if (callback) {
+            callback();
+          } else {
+            process.exit(0);
+          }
+        }
+      }
+    );
+  });
+};
+
 const program = new Command();
 program.argument("<sourceDir>").requiredOption("-o --output <value>");
 
 program.action(async (sourcePath: string, { output }) => {
-  const spinner = ora("generating components").start();
+  const stats = fs.lstatSync(sourcePath);
   const cwd = process.cwd();
   const outputPath = output.match(/^(\~|\/)/) ? output : path.join(cwd, output);
   sourcePath = sourcePath.match(/^(\~|\/)/)
     ? sourcePath
     : path.join(cwd, sourcePath);
 
-  const files = await getFiles(sourcePath);
-  const filesIterator = files[Symbol.iterator]();
-
-  const iterate = () => {
-    const filePath = filesIterator.next().value;
+  if (stats.isFile()) {
+    const filePath = getFile(sourcePath);
+    parseToJSXComponent(filePath, outputPath);
+  }
 
-    if (!filePath) {
-      spinner.stop();
-      process.exit(0);
-    }
+  if (stats.isDirectory()) {
+    const files = await getFiles(sourcePath);
+    const filesIterator = files[Symbol.iterator]();
 
-    const componentName = _upperFirst(
-      _camelCase(path.basename(filePath, ".svg"))
-    );
-    parser(filePath, componentName).then((text) => {
-      fs.mkdirSync(outputPath, { recursive: true });
-      fs.writeFile(
-        path.join(outputPath, componentName + ".tsx"),
-        text,
-        {
-          encoding: "utf-8",
-        },
-        (error) => {
-          if (error) {
-            console.warn(`error converting: ${filePath}`);
-          } else {
-            spinner.text = `${componentName + ".tsx"} generated`;
-          }
-          iterate();
-        }
-      );
-    });
-  };
+    const iterate = () => {
+      const filePath = filesIterator.next().value;
+      parseToJSXComponent(filePath, outputPath, iterate);
```

Comment:
and update spinner in here `spinner.text = `${filePath} converted successfully`;`